docs/specs/app-specs.md

```diff
@@ -1,121 +1,186 @@
 
-Libs specified in the depends dict have a different meaning.  They tell Dusty that these libs (and their recursive dependencies) need to be mounted into this app's container (using Docker's volume mounts).  In so doing, the running app will have access to the most recent code in these libs.
+`depends` is used to specify what libs, apps and services this app depends on.
+
+Containers for apps and services specified in the `depends` dict are created before the container
+for the referencing app. The referencing app is then linked (using [Docker links](https://docs.docker.com/userguide/dockerlinks/))
+to the dependent apps and services. Once containers are linked, they may reference each other through
+the environment variables or `/etc/hosts` overrides provided by Docker links.
+
+**N.B.:** All container links, including those from both `depends` and `conditional_links`, must be acyclical.
+
+Dependent libs (and their lib dependencies) will be mounted into the running app container and
+installed according to their specs.
+
+## conditional_links
 
-### conditional_links
 ```
 conditional_links:
   apps:
-    - app2
+    - myConditionalApp
 ```
-The conditional_links key is similar to the depends key. There are some big differences though. First, the conditional_links key can only specify apps. Second, the current app will only link to the apps specified (here app2) when: <br/>
-1. It is specified in different activated app's depends dict. <br/>
-2. It is specified in an activated bundles apps list. <br/>
-As in the depends dict, circular dependencies are not allowed.
 
-### host_forwarding
+`conditional_links` specifies downstream apps which are not required by the referencing app but
+should be linked if they are running as the result of a different bundle or dependency. This lets
+you layer optional containers onto a stack without forcing them all to be run whenever the main
+container is run.
+
+Unlike `depends`, `conditional_links` do not support links to libs.
+
+**N.B.:** All container links, including those from both `depends` and `conditional_links`, must be acyclical.
+
+## host_forwarding
+
 ```
 host_forwarding:
   - host_name: local.website.com
     host_port: 80
     container_port: 80
 ```
-The host_forwarding key is used specify a list of dictionaries.  Each dictionary allows you to specify a mapping from the containers ip space to your Mac's ip space.<br />
-**host_name** is the local url you would like to hit.<br />
-**host_port** is the port on which you will hit the host_name.<br />
-**container_port** is the port the app is running on in the Docker container.<br />
 
-### build
+`host_forwarding` allows you to specify multiple routes which should be linked between your host OS
+and the application running inside this app's container.
```

Comment:
URLs have schemes, so that's inaccurate. Really it's exactly what the spec says, a hostname and a port. Not sure what the best noun is for that.